File: Problems/20-Optimal-Partition-of-String/Optimal-Partition-of-String.cpp
class Solution {
public:
    int partitionString(string s) {
        int ans = 0;
        unordered_set<char> st;
        for (auto& c: s)
        {
            if (st.find(c) == st.end())
            {
                st.insert(c);
            }
            else
            {
                st.clear();
                st.insert(c);
                ans++;
            }
        }
        if (!st.empty()) ans++;
        return ans;
    }
};

File: Problems/20-Optimal-Partition-of-String/Optimal_Partition_of_String.py
class Solution(object):
    def partitionString(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 1
        tmp = []
        for char in s:
            if char in tmp:
                tmp = [char]
                res +=1
            else:
                tmp.append(char)
        return res
